Add a demo for JinnTap

DrRataplan opened the pull request:
Resolves #325 
* Prevent causing a refresh from immediately causing a setValue
* Do not change the value of the control if it would do nothing
* Add a demo page for jinntap
* Fix loading XML instances that load with `text/xml` mimetype


TODO: an E2E test for all this. This is quite finnicky

DrRataplan committed:
start working on jinntap demo

DrRataplan committed:
Fix fx-control controling an XML document

* Prevent causing a refresh from immediately causing a setValue
* Do not change the value of the control if it would do nothing
* Add a demo page for jinntap
* Fix loading XML instances that load with `text/xml` mimetype

JoernT reviewed (changes requested):
just a naming question....

DrRataplan committed:
Rename as="text" to as="xml"

fx-control as="text" would pass the serialized XML of an XML node to the widget. Name this as="xml".

Technically breaking, but since as="text" has been broken for years now, that's a pretty bnig signal
it's not being used at all

DrRataplan committed:
Move jinn-tap dep from main project to demo

DrRataplan commented on package.json:
```diff
@@ -38,6 +38,7 @@
   "devDependencies": {
     "@babel/plugin-proposal-class-properties": "^7.17.12",
     "@jinntec/jinn-codemirror": "^1.17.6",
+    "@jinntec/jinntap": "^1.27.2",
```
Moved

DrRataplan committed:
Mention jinn-tap demo on demos page

DrRataplan committed:
add missing example-2 file

DrRataplan committed:
Add E2E test for jinntap
